### PR TITLE
Improved support for different db drivers

### PR DESCRIPTION
**Pull request by core23**
I am targeting this branch, because this is BC.

## Changelog

```markdown
### Added
- Added support for flex

### Changed
- Improved support for different db drivers
- The default driver is now `no_driver`
```

## To do
- [x] Update the documentation
<s>- [ ] Create `news-odm-pack`</s>
<s>- [ ] Create `news-orm-pack`</s>

## Subject

This are the first steps for flex support.

### CODE REVIEW

**Comment by OskarStark** [comment]
Anything todo here @core23 ?

File: src/Resources/config/doctrine_mongodb.xml [hunk 1]
```diff
@@ -2,8 +2,9 @@
 <container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
     <parameters>
         <parameter key="sonata.news.manager.comment.class">Sonata\NewsBundle\Document\CommentManager</parameter>
-        <parameter key="sonata.news.manager.comment.entity">Application\Sonata\NewsBundle\Document\Comment</parameter>
         <parameter key="sonata.news.manager.post.class">Sonata\NewsBundle\Document\PostManager</parameter>
+        <!-- NEXT_MAJOR: Remove this keys -->
```
Comment (greg0ire) [inline src/Resources/config/doctrine_mongodb.xml]: Key

File: tests/Model/NoDriverManagerTest.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * @dataProvider providerMethods
+     */
+    public function testException($method, array $arguments)
+    {
+        $this->expectException(NoDriverException::class);
+        \call_user_func_array([new NoDriverManager(), $method], $arguments);
+    }
+
+    public function providerMethods()
```
Comment (greg0ire) [inline tests/Model/NoDriverManagerTest.php]: Either `provideMethods` or `methodsProvider` IMO

File: src/Resources/config/doctrine_mongodb.xml (hunk 1 above)
Comment (core23) [reply src/Resources/config/doctrine_mongodb.xml]: `keys` is correct in this case

File: tests/Model/NoDriverManagerTest.php (hunk 2 above)
Comment (core23) [reply tests/Model/NoDriverManagerTest.php]: It's just a copy and past 
https://github.com/sonata-project/SonataMediaBundle/blob/3.x/tests/Model/NoDriverManagerTest.php#L32

**Comment by OskarStark** [comment]
Please rebase

File: src/Exception/NoDriverException.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\NewsBundle\Exception;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ */
+class NoDriverException extends \RuntimeException
```
Comment (OskarStark) [inline src/Exception/NoDriverException.php]: final?

File: tests/Model/NoDriverManagerTest.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+use PHPUnit\Framework\TestCase;
+use Sonata\NewsBundle\Exception\NoDriverException;
+use Sonata\NewsBundle\Model\NoDriverManager;
+
+class NoDriverManagerTest extends TestCase
+{
+    /**
+     * @dataProvider mangerMethods
+     */
+    public function testException($method, array $arguments)
```
Comment (OskarStark) [inline tests/Model/NoDriverManagerTest.php]:
```suggestion
    public function testException(string $method, array $arguments)
```

File: src/Exception/NoDriverException.php (hunk 3 above)
Comment (core23) [reply src/Exception/NoDriverException.php]: Should we do this for exceptions?

Comment (OskarStark) [reply src/Exception/NoDriverException.php]: just an idea, but lets keep it for now

File: tests/Model/NoDriverManagerTest.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+namespace Sonata\NewsBundle\Tests\Model;
+
+use PHPUnit\Framework\TestCase;
+use Sonata\NewsBundle\Exception\NoDriverException;
+use Sonata\NewsBundle\Model\NoDriverManager;
+
+class NoDriverManagerTest extends TestCase
+{
+    /**
+     * @dataProvider mangerMethods
```
Comment (OskarStark) [inline tests/Model/NoDriverManagerTest.php]:
```suggestion
     * @dataProvider managerMethods
```

File: tests/Model/NoDriverManagerTest.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    public function testException(string $method, array $arguments)
+    {
+        $this->expectException(NoDriverException::class);
+        \call_user_func_array([new NoDriverManager(), $method], $arguments);
+    }
+
+    public function mangerMethods()
```
Comment (OskarStark) [inline tests/Model/NoDriverManagerTest.php]:
```suggestion
    public function managerMethods()
```

**Comment by core23** [comment]
Please review again @OskarStark

**Comment by greg0ire** [comment]
Should we merge this or wait for the todo list to be complete?

**Comment by core23** [comment]
The To Do list is optional. We don't have a flex recipe yet

**Comment by core23** [comment]
Can we merge this @greg0ire ?

Btw. we need a recipe for the user bundle before creating the news recipes

**Comment by OskarStark** [comment]
IIRC it’s not possible to create a recipe for the user bundle

**Comment by core23** [comment]
So we should find a way to create a recipe for this bundle (and the timeline bundle)

File: tests/Model/NoDriverManagerTest.php [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+class NoDriverManagerTest extends TestCase
+{
+    /**
+     * @dataProvider managerMethods
+     */
+    public function testException(string $method, array $arguments)
+    {
+        $this->expectException(NoDriverException::class);
+        \call_user_func_array([new NoDriverManager(), $method], $arguments);
```
Comment (phansys) [inline tests/Model/NoDriverManagerTest.php]:
```suggestion
        (new NoDriverManager())->{$method}(...$arguments);
```

Comment (phansys) [reply tests/Model/NoDriverManagerTest.php]: Could we avoid the call to `call_user_func_array()`?

**Pull request merged**